prisma/schema.prisma

```diff
@@ -18,4 +17,121 @@ model User {
   email     String   @unique
   name      String?
   createdAt DateTime @default(now())
+}
+// ============================================
+// MÓDULO TESTS
+// ============================================
+
+enum QuestionType {
+  MCQ
+  PROGRAMMING
+}
+
+enum DifficultyLevel {
+  EASY
+  MEDIUM
+  HARD
+}
+
+model Company {
+  id        String   @id @default(uuid())
+  name      String
+  logo      String?
+  tests     Test[]
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  @@map("companies")
+}
+
+model Test {
+  id          String          @id @default(uuid())
+  name        String
+  description String
+  company     Company         @relation(fields: [companyId], references: [id])
+  companyId   String
+  difficulty  DifficultyLevel
+  duration    Int
+  isActive    Boolean         @default(true)
+  questions   Question[]
+  sessions    TestSession[]
+  submissions Submission[]
+  createdAt   DateTime        @default(now())
+  updatedAt   DateTime        @updatedAt
+
+  @@index([companyId])
+  @@index([difficulty])
+  @@index([isActive])
+  @@map("tests")
+}
+
+model Question {
+  id              String          @id @default(uuid())
+  test            Test            @relation(fields: [testId], references: [id], onDelete: Cascade)
+  testId          String
+  type            QuestionType
+  text            String
+  difficulty      DifficultyLevel
+  points          Int
+  timeLimit       Int?
+  options         Json?
+  programmingData Json?
+  order           Int
+  createdAt       DateTime        @default(now())
+  updatedAt       DateTime        @updatedAt
+
+  @@index([testId])
+  @@index([type])
+  @@map("questions")
+}
+
+model TestSession {
+  id         String      @id @default(uuid())
+  test       Test        @relation(fields: [testId], references: [id], onDelete: Cascade)
+  testId     String
+  userId     String
+  startedAt  DateTime    @default(now())
+  expiresAt  DateTime
+  isActive   Boolean     @default(true)
+  answers    Answer[]
+  submission Submission?
+  createdAt  DateTime    @default(now())
+  updatedAt  DateTime    @updatedAt
+
+  @@index([testId, userId])
+  @@index([expiresAt])
+  @@index([isActive])
+  @@map("test_sessions")
+}
```

Comment:
Answer model declares questionId: String but does not establish a relation with the Question model. This means Prisma will not enforce referential integrity or provide the related object when querying.

**Sugerencia:**
```typescript
Add a relation field for the question, e.g.:
  question     Question   @relation(fields: [questionId], references: [id], onDelete: Cascade)
  questionId   String
and keep @@index([questionId]) as present. This mirrors the session->test relationship and ensures referential integrity. Priority: important
```